### PR TITLE
Update docker image, pip3 remove MPI

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- @geoffrey4444 requested having pip3 available inside the Docker image
- While getting Singularity containers set up on @geoffrey4444's cluster `Ocean` we noticed that OpenMPI was installed inside the container and Charm++'s configure script refused to find anything other than the container MPI. In order to get jobs inside the container to launch we had to compile Charm++ with the system MPI but inside the container. It turns out that `libboost-all-dev` on Ubuntu installs Boost.MPI, which caused OpenMPI to be installed. We now install the Boost libs we use (more can be installed if necessary).
- The number of cores to build on was hard coded in a few places in the Dockerfile. We replace them with the env variable that specifies it instead.

### Types of changes:

- [x] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [x] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: containers/Dockerfile.buildenv [hunk 1]
```diff
@@ -24,6 +24,13 @@ RUN apt-get update -y \
     && add-apt-repository ppa:ubuntu-toolchain-r/test
 
 # Install required packages for SpECTRE
+#
+# We intentionally don't install libboost-all-dev because that install
```
Comment (nilsvu) [inline containers/Dockerfile.buildenv]: `install` -> `installs`

File: containers/Dockerfile.buildenv [hunk 2]
```diff
@@ -55,7 +65,7 @@ RUN apt-get update -y \
 RUN apt-get install -y ccache doxygen
 
 # Install Python packages
-RUN apt-get install -y python-pip \
+RUN apt-get install -y python-pip python3-pip \
```
Comment (nilsvu) [inline containers/Dockerfile.buildenv:68]: while you're rebuilding the container, you might as well add the py3 version of the packages that are installed for py2 here. Just duplicate the lines and replace `pip` with `pip2` and `pip3`.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: containers/Dockerfile.buildenv (hunk 2 above)
Comment (nilsdeppe) [reply containers/Dockerfile.buildenv:68]: I don't want to make the image even larger... It's already huge and using up a lot of disk space. I'd rather have people install those packages themselves via pip or anaconda

File: containers/Dockerfile.buildenv (hunk 1 above)
Comment (nilsdeppe) [reply containers/Dockerfile.buildenv]: fixup pushed. Thanks!

File: containers/Dockerfile.buildenv (hunk 2 above)
Comment (nilsvu) [reply containers/Dockerfile.buildenv:68]: we'll need them to test on py2 and py3 on CI (numpy & scipy at least)

Comment (nilsdeppe) [reply containers/Dockerfile.buildenv:68]: Yes, I'll add numpy and scipy, but I don't think we should have all the others in the container. (We can remove later)

Comment (nilsvu) [reply containers/Dockerfile.buildenv:68]: fine with me for now, but we'll probably have to reconsider at some point what we want to have in the container and what we don't, or if we want to publish multiple containers.

**Comment by nilsdeppe** [comment]
@nilsleiffischer pushed fixups. The container is live on Dockerhub.

**Review by nilsvu — COMMENTED** [review]
LGTM, plese squash

**Comment by nilsvu** [comment]
While you're rebuilding the container, would you mind cherry-picking this commit 0f5341e69b07cda45f10c9bddf1344030d062cb7)? It's on my `nilsleiffischer/container_texlive` branch. I can also open a PR with it if you prefer, but it's probably faster to include it in this PR.

**Review by nilsvu — APPROVED** [review]

**Comment by nilsvu** [comment]
LGTM. Thanks for updating the container :)

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@geoffrey4444 please confirm this does what you want

**Comment by geoffrey4444** [comment]
@nils is this version any different from the newest one I pulled from dockerhub last week? If so I’ll rebuild the image and make sure all still works. Otherwise I approve

**Comment by nilsdeppe** [comment]
It also has numpy and scipy in it for python 3. Otherwise it's the same

**Review by geoffrey4444 — APPROVED** [review]

**Pull request merged**